eam-light-backendejb/src/main/java/ch/cern/cmms/eamlightejb/index/IndexGrids.java

```diff
@@ -88,16 +113,20 @@ private List<IndexResult> searchParts(InforContext inforContext, String keyword,
     }
 
     public List<IndexResult> search(InforContext inforContext, String keyword, List<String> entityTypes) throws InforException {
+        return search(inforContext, keyword, entityTypes, null, null);
+    }
+
+    public List<IndexResult> search(InforContext inforContext, String keyword, List<String> entityTypes, String entityClass, Integer rowCount) throws InforException {
         List<IndexResult> result = new LinkedList<>();
         List<Runnable> runnables = new LinkedList<>();
 
         Map<String, Runnable> entityTypeRunnableMap = new HashMap();
-        entityTypeRunnableMap.put("JOB", () -> result.addAll(searchWorkOrders(inforContext, keyword,"BEGINS")));
-        entityTypeRunnableMap.put("A", () -> result.addAll(searchEquipment(inforContext, keyword,"BEGINS","OSOBJA","A", true)));
-        entityTypeRunnableMap.put("P", () -> result.addAll(searchEquipment(inforContext, keyword,"BEGINS","OSOBJP","P", true)));
-        entityTypeRunnableMap.put("S", () -> result.addAll(searchEquipment(inforContext, keyword,"BEGINS","OSOBJS","S", true)));
-        entityTypeRunnableMap.put("L", () -> result.addAll(searchEquipment(inforContext, keyword,"BEGINS","OSOBJL","L", false)));
-        entityTypeRunnableMap.put("PART", () -> result.addAll(searchParts(inforContext, keyword, "BEGINS")));
+        entityTypeRunnableMap.put("JOB", () -> result.addAll(searchWorkOrders(inforContext, keyword, "BEGINS", rowCount - result.size())));
+        entityTypeRunnableMap.put("A", () -> result.addAll(searchEquipment(inforContext, keyword, "BEGINS", "OSOBJA", "A", true, entityClass, rowCount - result.size())));
+        entityTypeRunnableMap.put("P", () -> result.addAll(searchEquipment(inforContext, keyword, "BEGINS", "OSOBJP", "P", true, entityClass, rowCount - result.size())));
+        entityTypeRunnableMap.put("S", () -> result.addAll(searchEquipment(inforContext, keyword, "BEGINS", "OSOBJS", "S", true, entityClass, rowCount - result.size())));
+        entityTypeRunnableMap.put("L", () -> result.addAll(searchEquipment(inforContext, keyword, "BEGINS", "OSOBJL", "L", false, entityClass, rowCount - result.size())));
```

Comment:
We should take this opportunity to search for locations (L) first, since the codes are usually shorter and often match the alias/serial number of assets